pytorch_lightning/accelerators/ddp_accelerator.py

```diff
@@ -169,7 +167,7 @@ def _step(self, args):
         return output
 
     def barrier(self, name: Optional[str] = None):
-        if torch_distrib.is_initialized():
+        if torch_distrib.is_initialized() and self.ddp_plugin.use_barrier_and_broadcast:
```

Comment:
Let's try get this check unneeded. From what I understand, we just have to ensure that the groups are correct (ensure that we communicate to all processes within the data parallel group).